erc20/src/lib.rs

```diff
@@ -67,10 +78,16 @@ impl ERC20 {
         self.allowances.read(owner).read(spender)
     }
 
-    pub fn mint(&self, to: Address, value: u64) {
```

Comment:
Remove these changes right?

And can add corresponding transfer logs

Reply:
I don't know which is the final version since the previous mint function had a different logic.

Which one should we go with?


```rust
    pub fn mint(&self, to: Address, value: u64) {
        let owner = msg_sender();
        let to_balance = self.balances.read(to);
        self.balances.write(to, to_balance + value);
    }
```
    
```rust
        #[payable]
    pub fn mint(&self, to: Address, value: u64) -> bool {
        let owner = msg_sender();
        if owner != address!("0000000000000000000000000000000000000001") {
            revert();
        }

        let to_balance = self.balances.read(to);
        self.balances.write(to, to_balance + value);

        true
    }
```

Reply:
add :

```rust
#[derive(Event)]
pub struct Mint {
    #[indexed]
    pub from: Address,
    #[indexed]
    pub to: Address,
    pub value: u64
}

    #[payable]
    pub fn mint(&self, to: Address, value: u64) -> bool {
        let owner = msg_sender();
        if owner != address!("0000000000000000000000000000000000000001") {
            revert();
        }

        let to_balance = self.balances.read(to);
        self.balances.write(to, to_balance + value);
        log::emit(Mint::new(owner, to, value));
        true
    }
```